lambda_backend/predict_microservice/base_inference.py

```diff
@@ -0,0 +1,144 @@
+import cv2
+import numpy as np
+from configparser import RawConfigParser
+
+DEF_SCORE = 0.3
+DEF_NMS = 0.4
+
+class BaseInference:
+    """
+    Base Inference class for predictions
+    
+    ...
+
+    Attributes
+    ----------
+    weight_path : str
+        path to the .weight file
+    config_path : str
+        path to the .cfg file
+    classes : list
+        names of classes detected
+    score_thresh : float
+        threshold to classify object as detected
+    nms_threshold : float
+        threshold for non-max suppression
+
+    Methods
+    -------
+    initialize()
+        Initializes neural network using given weight and config
+    read_config()
+        Reads config for trained height and width
+    run()
+        Obtains predicted boxes 
+    """
+
+    def __init__(self, weight_path, config_path, classes, score_thresh=None, nms_thresh=None):
+        self.weight_path = weight_path
+        self.config_path = config_path
+        self.classes = classes
+        self.net = None
+        self.score_thresh = score_thresh if score_thresh is not None else DEF_SCORE
+        self.nms_thresh = nms_thresh if nms_thresh is not None else DEF_NMS
+    
+    def initialize(self):
+        # Load Yolo
+        self.net = cv2.dnn.readNet(
+            self.weight_path,
+            self.config_path
+        )
+        layer_names = self.net.getLayerNames()
+        self.output_layers = [layer_names[i[0] - 1] for i in self.net.getUnconnectedOutLayers()]
+    
+    def read_config(self):
+        cfg = RawConfigParser(strict=False)
+        cfg.read(self.config_path)
+
+        net_dict = dict(cfg.items('net'))
+        self.train_height_width = (int(net_dict['height']), int(net_dict['width']))
+
+    def run(self, img):
+        """
+        Parameters
+        ----------
+        img : cv2.Mat
+            Image as a matrix
+        
+        Returns
+        -------
+        class_ids : list(int)
+            Class IDs of boxes
+        box_dims : list(list(int))
+            Dimensions of boxes
+        box_confidences : list(float)
+            Confidence scores of boxes
+        dets : list(list(float))
+            Normalised dimensions of boxes
+        indexes : list(int)
+            Indexes of boxes that passed NMS
+        """        
```

Comment:
Can add validation to check if `img` is of the correct data type, `numpy.ndarray`?